.github/workflows/image-canary.yaml

```diff
@@ -0,0 +1,60 @@
+name: ImageCanary
+on:
+  schedule:
+  - cron: '0 */1 * * *'
+jobs:
+  ImageCanary:
+    runs-on: ubuntu-latest
+    permissions:
+      id-token: write # aws-actions/configure-aws-credentials@v4.0.1
+    if: github.repository == 'aws/karpenter-provider-aws'
+    steps:
+    - uses: actions/checkout@9bb56186c3b09b4f86b1c65136769dd318469633 # v4.1.2
+    - name: Configure AWS credentials
+      uses: aws-actions/configure-aws-credentials@e3dd6a429d7300a6a4c196c26e071d42e0343502 # v4.0.2
+      with:
+        role-to-assume: arn:aws:iam::${{ vars.READONLY_ACCOUNT_ID }}:role/${{ vars.READONLY_ROLE_NAME }}
+        aws-region: ${{ vars.READONLY_REGION }}
+        role-duration-seconds: 900
+    # Authenticate to public ECR to prevent rate limiting
+    - uses: docker/login-action@e92390c5fb421da1463c202d546fed0ec5c39f20 # v3.1.0
+      with:
+        registry: public.ecr.aws
+        logout: true
+      env:
+        AWS_REGION: us-east-1
+    - name: Pull artifacts
+      env:
+        GH_TOKEN: ${{ github.token }}
+      run: |
+        # Translates a git tag to the corresponding ECR tag. Starting with Karpenter v0.35.0 the leading v is dropped in the ECR tag.
+        function getECRTag() {
```

Comment:
What's the trade-off here between the maintainability of the go script that you had written and this one? Do we expect to need to make a ton of changes to this script file in the future? If so, do we expect others on the team to easily be able to make modifications and test this code?

Reply:
I didn't rebase out the go version so you can take a look at the previous commit if you want. This version strips some of the functionality of the go version away. The go version basically wrapped the GitHub API and output json array of the versions, with an optional filter for the n latest minor versions. 
```json
[
  {
    "version": {
      "major": 0,
      "minor": 35,
      "patch": 0
    },
    "gitTag": "v0.35.0",
    "ecrTag": "0.35.0"
  },
  ...
]
```
Everything here from line 42 down remained the same. I think for this single application, this is the way to go since there's no real esoteric bash (except maybe Sed's flavor of regex). It could make sense to go the go route if we think we could make use of the script elsewhere as well (I was thinking it could be useful in docgen).